domain/consensus/factory.go

```diff
@@ -418,6 +420,9 @@ func (f *factory) NewConsensus(config *Config, db infrastructuredatabase.Databas
 	}
 
 	if !genesisInfo.Exists {
+		if c.blockStore.Count(model.NewStagingArea()) > 0 {
+			return nil, errors.Errorf("Genesis hash is not relevant to database: wrong config or appdir?")
```

Comment:
I'd be more implicit with the problem... something like:
"Expected genesis block {genesis hash} for {Params.name} is not found in the DAG, while the DAG is not empty".

Also note: you are using errors.Errorf(), with no formatting token in it. In such cases better use errors.New() (but since what I said earlier, there's going to be format tokens now.)